x-pack/plugins/security_solution/public/common/components/url_state/helpers.ts

```diff
@@ -141,6 +187,23 @@ export const makeMapStateToProps = () => {
           }
         : { id: '', isOpen: false, activeTab: TimelineTabs.query, graphEventId: '' };
 
+    const timelineId = getTimelineIdForPathname(ownProps?.pathName, flyoutTimeline?.show);
+    let detailPanel = {};
+
+    if (timelineId) {
+      const activeTimelineById =
```

Comment:
if you move this logic up, you could do this at the same time as flyoutTimeline is created, i.e. only call getTimeline once

Reply:
I actually would need to leave it. The timelines are doing different things. One is always requesting the flyout timeline, while this one is requesting the one for whichever alert table is associated with your current path. Since this is looking for the topmost details panel, it defaults to the already called `getTimeline` if the top most one happens to be the flyout, else it gets the other one